basic_block_gp/blockchain.py

```diff
@@ -151,19 +162,27 @@ def valid_chain(self, chain):
 blockchain = Blockchain()
 
 
-@app.route('/mine', methods=['GET'])
+@app.route('/mine', methods=['GET', 'POST'])
 def mine():
     # We run the proof of work algorithm to get the next proof...
-    proof = blockchain.proof_of_work()
+    
+    data = request.json
+    print(f'{data} AAAAAAAAAA')
+    proof = blockchain.proof_of_work(blockchain.last_block)
 
     # We must receive a reward for finding the proof.
     # TODO:
     # The sender is "0" to signify that this node has mine a new coin
     # The recipient is the current node, it did the mining!
     # The amount is 1 coin as a reward for mining the next block
-
+    blockchain.new_transaction(
+        sender="0",
+        recipient=node_identifier,
+        amount=1,
+        )
     # Forge the new Block by adding it to the chain
-    # TODO
+    previous_hash = blockchain.hash(blockchain.last_block)
+    block = blockchain.new_block(data, previous_hash)
```

Comment:
you're adding a block every time you receive a request...sometimes you wouldn't want to add the new block, when to check and what to check for?